### DISCUSSION TITLE
HttpEndpoint keep showing error: "The call is ambiguous and matches multiple workflows."

### DISCUSSION THREAD

**Discussion by kfconeone** [elsa-workflows/elsa-core#6172] [Q&A] [answered]
Hi guys,

I’m a beginner with Elsa and not very familiar with its workflow engine yet. I’ve spent an entire week diving into it day and night, but I still don’t fully understand how it works. I’m currently stuck on an issue and would like to ask for your help:
![image](https://github.com/user-attachments/assets/933e6002-d060-428a-a0b5-8ab5c8c9d5b3)
![image](https://github.com/user-attachments/assets/5a90e282-1902-47d4-bf07-b49223f15ec2)

As shown in the image above, I’m trying to achieve the behavior illustrated. In my POST request body, I’ll pass a boolean value, and based on this value, I want to determine what response is returned. I have only one workflow, but there are two instances stuck at this point.

The result keeps showing:
"errorMessage": "The call is ambiguous and matches multiple workflows."

Am I misunderstanding something? Please help me out.

Also, I’d like to ask if there are any learning resources available for Elsa, whether books or online courses. I’ve had a hard time finding anything 😅. Many thanks!

**Comment by Suchiman** [comment]
I don't think that it makes sense, to have {CorrelationId} in the Url of an HttpEndpoint, that is set to trigger the workflow.
Also make sure that this is the only workflow definition, that uses that particular url.

**Reply by kfconeone** [reply]
I'm pretty sure I have only one definition, and because of the word "ambiguous ", make me think I should have some id to identify the certain instance of my workflow, whether in the Url or request body.

So, if this idea doesn't make sense, what should I do to distinguish my specific instance without triggering ambiguous call?

**Reply by Suchiman** [reply, marked as answer]
It's fine to have a "non unique" / hardcoded url for the trigger of the workflow, as long as there's only one workflow waiting for it.
But, relaying from discord here, make sure there's no manually started instance of this workflow. If you manually start it, it would then suspend at the HttpEndpoint activity and then you have a workflow that wants to trigger on some Url and another workflow that has a bookmark on the same Url, which could cause this issue. So best to go to instances and remove all of them.

**Reply by kfconeone** [reply]
Wow, thanks for your reply, I finally figured it out!
